pytest_operator/plugin.py

```diff
@@ -135,17 +140,6 @@ def pytest_runtest_setup(item):
         pytest.skip("Skipping deployment because --no-deploy was specified.")
 
 
-@pytest.fixture(scope="session")
-def tmp_path_factory(request):
-    # Override temp path factory to create temp dirs under Tox env so that
-    # confined snaps (e.g., charmcraft) can access them.
-    return pytest.TempPathFactory(
-        given_basetemp=Path(os.environ["TOX_ENV_DIR"]) / "tmp" / "pytest",
-        trace=request.config.trace.get("tmpdir"),
-        _ispytest=True,
-    )
-
-
```

Comment:
this was such a pain to unit test, it was so much easier to replace that using the `basetemp` configuration in pytest